Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1158,6 +1166,88 @@ extension BrowserViewController: AppStateDelegate {
     }
 }
 
+extension BrowserViewController: MenuActionDelegate {
+    func performMenuAction(action: MenuAction, withAppState appState: AppState) {
+        switch action {
+        case .OpenNewNormalTab:
+            dispatch_async(dispatch_get_main_queue()) {
```

Comment:
I'd follow iOS's UI threading model here, which is that it's the caller's responsibility to ensure that any UI code is touched only on the main thread. Having async dispatches hidden in the method like this can lead to subtle bugs; for example, I call `performMenuAction` to open a new tab, but the number of tabs won't change right after this call. Generally, I think we're doing something wrong if we have background threads firing off menu actions!

For the sake of safety, it'd be good to add `assert(NSThread.isMainThread())` at the top of the method (like we do in `TabManager`) so we can catch cases where we violate this constraint.